README.md

```diff
@@ -1 +1,308 @@
+- [Usage](#usage)
+- [Examples](#examples)
+- [Contributing](#contributing)
+- [License](#license)
+
+
+## Purpose
+
+1.	Connect customers with cooks who offer homemade food.
+2.	Support local cookers and help customers who are looking for homemade food
+
+
+
+## Features
+
+1.	Connects customers with local cooks offering homemade food.
+2.	Browse and explore a diverse range of homemade dishes.
+3.	User-friendly interface for placing and managing orders.
+4.	Profiles for cooks to showcase their culinary offerings.
+5.	Order tracking and communication between customers and cooks.
+
+## Installation
+
+To install ChatGPT Devs Era, you can use yarn:
+
+```bash
+yarn install chatgpt-devs-era
+```
+
+### To start the server
+
+run the command `yarn && yarn start`. This will install all the dependencies
+
+### To run the tests
+
+run the command `yarn test`. Make sure that the server is up and running as well as the database
+
+### To install packages
+
+when you run `yarn add PACKAGE` the package will be installed in docker as well automatically. However if you run into issues, you need to stop the server first with `yarn run stop` then you can run `yarn run build` to rebuild the docker image and start again.
+
+## Usage
+
+To interact with the backend server, you can make HTTP requests to the following endpoints. The base URL for all endpoints is http://localhost:3000/api.
+
+Endpoints
+1. Create Normal User (Signup)
+Description: Creates a new normal user account.
+Endpoint: /auth/signup
+Method: POST
+Request Body:
+```json
+{
+    "firstname": "",
+    "lastname": "",
+    "email": "",
+    "password": "",
+    "role": ""
+}
+```
+
+2. Login Normal User
+Description: Authenticates the normal user and generates an access token for subsequent requests.
+Endpoint: /auth/login
+Method: POST
+Request Body:
+```json
+{
+    "email": "",
+    "newPassword": ""
+}
+```
+
+3. Password Reset
+Description: Initiates the password reset process for a user.
+Endpoint: /auth/passwordReset
+Method: POST
+Request Body:
+```json
+{
+    "email": "",
+    "password": ""
+}
+```
+4. Logout
+Description: Logs out the currently logged-in user and invalidates the access token.
+Endpoint: /auth/logout
+Method: POST
+
+5. Decide User Role
+Description: Allows the system to assign a role to a user, based on certain criteria or user preferences.
+Endpoint: /role
+Method: PUT
+Request Body:
+```json
+{
+    "providerId": "",
+    "role": "client"
+}
+```
+
+
+6. Create Client Profile
+Description: Creates a new client profile.
+Endpoint: /clients/
+Method: POST
+Request Body:
+```json
+{
+  "address": "",
+  "contactNumber": ""
+}
+```
+
+7. Get Client Profile
+Description: Retrieves the profile of the authenticated client.
+Endpoint: /clients/me
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+8. Update Client Profile
+Description: Updates the profile of the authenticated client.
+Endpoint: /clients/me
+Method: PUT
+Request Body:
+```json
+{
+  "address": ""
+}
```

Comment:
we need to add contact number here also, cause it can be edited

Reply:
Oh really good @ozlemkeles, I didn't catch this one.